scipy/special/tests/test_basic.py

```diff
@@ -2152,14 +2154,21 @@ def test_factorial_array_corner_cases(self, content, dim, exact, dtype):
             # no error
             result = special.factorial(n, exact=exact)
 
+        # https://github.com/numpy/numpy/issues/24050
```

Comment:
```suggestion
        # assert_equal does not distinguish scalars and 0-dim arrays of the same value, see
        # https://github.com/numpy/numpy/issues/24050
```
Uncommitted comment improvement.